### ISSUE TITLE
Improve integration test suite speed.

### ISSUE THREAD

**Issue by chuckharmston** [mozilla/testpilot#2536]
Our build times have doubled in the last week:

- https://circleci.com/gh/mozilla/testpilot/4473
- https://circleci.com/gh/mozilla/testpilot/4503

And we even have builds that are occasionally timing out: https://circleci.com/gh/mozilla/testpilot/4539

This appears to be due to the addition of integration tests that test across three versions of Firefox. While good to have, I don't think that this is acceptable; I'd be in favor of backing these tests out until we can get builds to a more reasonable time.

**Comment by b4handjr** [comment]
So for the most part they aren't timing out but it seems every now and then the browser does not load. My current config adds 2 more workers to a total of 4. And helps increase the build time just a hair. 

I have been looking into circle's parallel builds but I have not really done much with them as of yet. I think the only quickest way of reducing times would be to disable all 3 versions of firefox from being tested in serial, and maybe figure out a way to run them in parallel.

**Comment by lmorchard** [comment]
Copying over some further scrollback from IRC:
```
<fzzzy> maybe we should just test nightly and release?
<lorchard> Well, that only saves about a minute & a half
<fzzzy> ah bummer
<lorchard> Not nothing, but they each take about 90 seconds between downloading the browser binary and firing everything up
<chuck> We should cache the binaries if we aren't doing that already, that will help a lot.
<lorchard> Probably some things we could do with caching the binaries, not sure about the rest
<chuck> Might make sense to run them as a separate nonblocking job.
<lorchard> we switched to mozdownload, but that doesn't seem to have caching out of the box
<chuck> Circle should be able to cache them.
<lorchard> But we have to know when to download new versions & cache invalidate
<lorchard> not simple file caching
<lorchard> I don't imagine it's impossible, just needs to be investigated
<lorchard> Can probably also run the tests in parallel, but not sure how off the top of my head
<lorchard> Could also just run them on master branch and speed up PR checks, though that kind of defeats the purpose
<fzzzy> lots of builds are timing out
<lorchard> Reticent to just back out the testing, because we've had a bad history of just turning off and never getting back around to dealing with integration test
<lorchard> But maybe we need to? meh
<_6a68> iafter working on Gecko for a while, where running automated tests takes like half a day, 8 minutes seems like, ¯\_(ツ)_/¯
<_6a68> *after
<lorchard> We should also run tests a lot less often if/when we move pontoon commits to an l10n branch
<b4hand> How about this? https://github.com/michaelcontento/circleci-matrix
<b4hand> Circleci doesn't supoort matrix's like travis
<lorchard> Yeah, that seems promising
```

**Comment by b4handjr** [comment]
Got a parallel build up to see how long it takes. https://circleci.com/gh/jrbenny35/testpilot/200#tests/containers/1

**Comment by b4handjr** [comment]
Running this on 4x parallelism would be worse it seems. We only have 1 container through circle and that supports up to 4 builds. Choosing 4x means we would have to wait until there is no one else building before it would start.

I just want to clarify that the tests themselves are not flaky. It appears to be that sometimes firefox just doesn't start, or the server doesn't ever serve the webpage.

**Comment by ghost** [comment]
We don't run tests on Pontoon commits.  We got more circle workers.  jrbenny35 has tests running on dev again.

@chuckharmston can this be closed?

**Comment by fzzzy** [comment]
We still have some issues with intermittent failures, but there is a new issue to look into that. After some discussion on irc with chuck, closing this.

https://github.com/mozilla/testpilot/issues/2625
